kernel/packages/decentraland-ecs/src/decentraland/math/Quaternion.ts

```diff
@@ -526,37 +527,54 @@ export class Quaternion {
    * @param up - defines the direction
    */
   public setFromToRotation(from: Vector3, to: Vector3, up: Vector3 = MathTmp.staticUp) {
-    MathTmp.tmpMatrix = Matrix.Zero() // clean up preallocated matrix
-    Matrix.LookAtLHToRef(from, to, up, MathTmp.tmpMatrix)
-    MathTmp.tmpMatrix.invert()
-    Quaternion.FromRotationMatrixToRef(MathTmp.tmpMatrix, this)
+    const result = Quaternion.FromToRotation(from, to)
+    this.x = result.x
+    this.y = result.y
+    this.z = result.z
+    this.w = result.w
   }
```

Comment:
Is OK for this not to use UP vector? rotation values shouldn't be different depending on UP vector?

Reply:
fuck, you are right, OK I'll see what I can do

Reply:
YES it should take into account the up vector.